### PR TITLE
TP-172 Fix hardcoded breadcrumb

### PR DESCRIPTION
**Pull request by DIT-CJWaszczuk**
## Summary 
Detail breadcrumb had hardcoded link text, made that dynamic using the same standard as list.

### CODE REVIEW

File: common/jinja2/layouts/detail.jinja [hunk 1]
```diff
@@ -1,10 +1,12 @@
 {% extends "layouts/layout.jinja" %}
 
+{% set object_type_plural = object_type_plural or object_type + "s" %}
+
 {% block breadcrumb %}
   {{ govukBreadcrumbs({
     "items": [
       {"text": "Home", "href": url("index")},
-      {"text": "Find and edit footnotes", "href": find_edit_url},
+      {"text": "Find and edit " ~ (object_type_plural or object_type + "s"), "href": find_edit_url},
```
Comment (NDevox) [inline common/jinja2/layouts/detail.jinja:9]: As you already set `object_type_plural` to use `s` if one doesn't exist you can just replace this line with `"Find and edit " ~ object_type_plural`

Comment (DIT-CJWaszczuk) [reply common/jinja2/layouts/detail.jinja]: I followed the same suit as the [list](https://github.com/uktrade/tamato/blob/master/common/jinja2/layouts/list.jinja) file (I think  we spoke about the necessity of it there but I can't remember why we chose to keep it). I've switched this one over, do you remember why we kept it there?

Comment (NDevox) [reply common/jinja2/layouts/detail.jinja]: No idea why we kept that 🤔  looks like it should be changed over too.

**Review by NDevox — APPROVED** [review]

File: common/jinja2/layouts/detail.jinja [hunk 2]
```diff
@@ -1,10 +1,12 @@
 {% extends "layouts/layout.jinja" %}
 
+{% set object_type_plural = object_type_plural or object_type + "s" %}
```
Comment (andyhd) [inline common/jinja2/layouts/detail.jinja:3]: I'd be tempted to make these available from the models, eg:
```python
class TrackedModel(...):
    ...
    @property
    def type_name(self) -> str:
        return self.__class__.__name__

    @property
    def type_name_plural(self) -> str:
        return f"{self.type_name}s"
```
and
```python
class GoodsNomenclature(TrackedModel, ...):
    ...
    @property
    def type_name(self) -> str:
        return "Commodity"
    
    @property
    def type_name_plural(self) -> str:
        return "Commodities"
```
Then you wouldn't need these `{% set ... %}` tags and could do
```jinja2
{{ "Find and edit " ~ object.type_name_plural }}
```

**Pull request merged**